python/pylibcudf/pylibcudf/libcudf/rolling.pxd

```diff
@@ -1,13 +1,210 @@
-# Copyright (c) 2020-2024, NVIDIA CORPORATION.
+# Copyright (c) 2020-2025, NVIDIA CORPORATION.
+from libcpp cimport bool
 from libcpp.memory cimport unique_ptr
+from libcpp.pair cimport pair
+from libcpp.vector cimport vector
 from pylibcudf.exception_handler cimport libcudf_exception_handler
-from pylibcudf.libcudf.aggregation cimport rolling_aggregation
+from pylibcudf.libcudf.aggregation cimport Kind, rolling_aggregation
 from pylibcudf.libcudf.column.column cimport column
 from pylibcudf.libcudf.column.column_view cimport column_view
-from pylibcudf.libcudf.types cimport size_type
+from pylibcudf.libcudf.scalar.scalar cimport scalar
+from pylibcudf.libcudf.table.table cimport table
+from pylibcudf.libcudf.table.table_view cimport table_view
+from pylibcudf.libcudf.types cimport data_type, null_order, order, size_type
 
 
 cdef extern from "cudf/rolling.hpp" namespace "cudf" nogil:
+    cdef cppclass rolling_request:
+        rolling_request() except +libcudf_exception_handler
+        column_view values
+        unique_ptr[rolling_aggregation] aggregation
```

Comment:
This is why the `rolling_request` doesn't take a reference to the `column_view` and reference to the rolling aggregation: the usual cython issues around a combination of aggregate initialisation vs. ctors, and being able to construct an "empty" struct.

Reply:
Presumably the problem is not that you cannot expose an API that accepts a reference, but rather that you cannot stack-allocate a `rolling_aggregation` in Cython to use via reference so you need to build it on the heap anyway?

Reply:
It's because you can't do:
```
rolling_aggregation& agg;
age = the_actual_agg;
```

Or something morally like that, which is a pattern cython loves to do.

Reply:
That is the usual problem with Cython, I was just clarifying that nothing in Cython prevents you from exposing the reference-based API here in rolling.pxd, right? The issue is that if you exposed that API you would still need to allocate the aggregation on the heap wherever you were going to call this API and then pass it via `obj.get()`?

Reply:
yes